src/Histogram.ts

```diff
@@ -175,4 +217,40 @@ export default class Histogram {
     }
     return [b, e];
   }
+
+  private calculateHistogram(arr: TypedArray<NumberType>, numBins = 1): HistogramData {
+    if (numBins < 1) {
+      numBins = 1;
+    }
+
+    // calculate min and max of arr
+    // TODO FIXME See convertChannel, which will also compute min and max!
+    // We could save a whole extra loop over the data, or have convertChannel compute the whole histogram.
+    // need to be careful about computing over chunks or whole ready-to-display volume
```

Comment:
Could be - it's not that anything's broken. More of a potential to optimize a bit more.  It would be great if we only have to loop over the data 0 or 1 times and the rest is just moving array refs around or copying to gpu.